### PR TITLE
regenerated with 2.21

### PR DESCRIPTION
**Pull request by miklossy**

### CODE REVIEW

File: languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java [hunk 1]
```diff
@@ -448,13 +446,7 @@ public ParserRule getXOtherOperatorExpressionRule() {
 	}
 	
 	//OpOther:
-	//	'->'
```
Comment (cdietrich) [inline languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java]: wonder why this has changed

Comment (cdietrich) [reply languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java]: i dont get these changes but changes to generated xtend code

Comment (miklossy) [reply languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java]: Do you mean that one line or the entire comment change?

**Comment by miklossy** [comment]
I did not regenerare the Xtend code base, just executed the MWE2 workflows.

File: languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java (hunk 1 above)
Comment (cdietrich) [reply languages/org.xtext.builddsl/src-gen/org/xtext/builddsl/services/BuildDSLGrammarAccess.java]: dont get any grammar access changes

**Comment by miklossy** [comment]
Are you aware of any code change in the Xtext framework itself that could cause this change? There have been sinilat changes in the Xtext examples as well: org.eclipse.xtext.xtext.ui.examples/projects/arithmetics/org.eclipse.xtext.example.arithmetics/src-gen/org/eclipse/xtext/example/arithmetics/services/ArithmeticsGrammarAccess.java

**Comment by cdietrich** [comment]
no. which change/commit do you mean?

**Comment by miklossy** [comment]
I mean [this commit](https://github.com/eclipse/xtext-eclipse/commit/2559050419669311f508783db5afdb0d23baeaa5#diff-38431c3e8ab606a3a5845e4e970d3d63) 
image

**Comment by miklossy** [comment]
or it could have something to do with the EMF version?

**Comment by cdietrich** [comment]
this change is about `(` and does not affect line breaks

**Comment by cdietrich** [comment]
no. but maybe with windows

**Comment by cdietrich** [comment]
but you have to look into the fragment to see what it does

**Comment by miklossy** [comment]
ah, ok. @ArneDeutsch Could you please regenerate the code base with Xtext 2.21 under Windows?

**Comment by ArneDeutsch** [comment]
I have checked out the examples with latest Xtext version from your branch. I executed `GenerateBuildDSL.mwe2` from eclipse. In `BuildDSLGrammarAccess.java` I have only some whitespace changes inside comments e.g.

```
//BuildFile:
//	("package" name=QualifiedName)? importSection=XImportSection? declarations+=Declaration*;
@Override public ParserRule getRule() { return rule; }
```
==>
```
//BuildFile:
//	("package" name=QualifiedName)?
//	importSection=XImportSection?
//	declarations+=Declaration*;
@Override public ParserRule getRule() { return rule; }
```

**Comment by cdietrich** [comment]
so you do see the same changes as tamas or do you see what is on master @ArneDeutsch

**Comment by ArneDeutsch** [comment]
I executed it again against master ... and there I see the same changes as reported my Tamas.

**Comment by cdietrich** [comment]
then we need to find out why this happens and why it does not on linux.

**Comment by miklossy** [comment]
I analyzed this problem a little bit and could find out that the problem does not appear when deactivating the formatting in the serialization in the [GrammarAccessExtension.grammarFragmentToString](https://github.com/eclipse/xtext-core/blob/master/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/grammarAccess/GrammarAccessExtensions.xtend#L413) method.

**Comment by miklossy** [comment]
So the problem could be further narrowed down that the Xtext formatter formats the Xbase grammar differently according to the used line endings. E.g when formatting the OpOther rule from the Xbase grammar with Linux line endings, there is no change, but formatting it with windows line endings, everything will be formatted into one line:

**Comment by miklossy** [comment]
The different formatting result occur when formatting an `*.xtext` file containing Linux line endings on Windows. The problem can be reproduced by the following unit test:
```
	@Test def void testGrammarFragmentToString() {
		TerminalsStandaloneSetup.doSetup
		val injector = Guice.createInjector(new XtextRuntimeModule(), new Module(){

			override configure(Binder binder) {
				binder.bind(IXtextProjectConfig).toInstance(new StandardProjectConfig)
			}

		});

		val grammarAccessExtension = injector.getInstance(GrammarAccessExtensions)
		val parserHelper = injector.getInstance(ParseHelper)


		var grammar = parserHelper.parse('''
		grammar org.xtext.example.mydsl.MyDsl

		generate myDsl "http://www.xtext.org/example/mydsl/MyDsl"

		OpOther:
			  '->'
			| '..<'
			| '>' '..'
			| '..'
			| '=>'
			| '>' (=>('>' '>') | '>')
			| '<' (=>('<' '<') | '<' | '=>')
			| '<>'
			| '?:';
		'''.toString.toUnixLineSeparator) as Grammar // Removing the toUnixLineSeparator makes the test pass

		val rule = grammar.rules.head

		val actual = grammarAccessExtension.grammarFragmentToString(rule, "//")
		Assert.assertEquals('''
		//OpOther:
		//	'->'
		//	| '..<'
		//	| '>' '..'
		//	| '..'
		//	| '=>'
		//	| '>' (=> ('>' '>') | '>') | '<' (=> ('<' '<') | '<' | '=>') | '<>'
		//	| '?:';
		'''.toString.trim, actual)
	}
```
@cdietrich Does this help us further?

**Comment by cdietrich** [comment]
No it does not fix the bug in the fragment

**Comment by miklossy** [comment]
I created a corresponding [issue](https://github.com/eclipse/xtext-core/issues/1443) on the xtext-core repository.

**Comment by miklossy** [comment]
I provided a fix https://github.com/eclipse/xtext-core/pull/1444 to this problem.

@ArneDeutsch Could you please regenerate the code with Xtext latest **on Windows** to verify that there is no undesired changes in the comments of the `GrammarAccess` classes?

**Pull request closed without merging**